lambdas/check-coi/src/main/java/uk/gov/di/ipv/core/checkcoi/CheckCoiHandler.java

```diff
@@ -212,15 +216,45 @@ private void sendAuditEvent(
             String ipvSessionId,
             String govukSigninJourneyId,
             String ipAddress)
-            throws SqsException {
+            throws SqsException, HttpResponseExceptionWithErrorBody, CredentialParseException,
+                    VerifiableCredentialException {
         var auditEventUser =
                 new AuditEventUser(userId, ipvSessionId, govukSigninJourneyId, ipAddress);
 
+        var restrictedData =
+                auditEventType == AuditEventTypes.IPV_CONTINUITY_OF_IDENTITY_CHECK_END
+                        ? getRestrictedAuditDataforCheckCoi(userId, ipvSessionId)
+                        : null;
+
         auditService.sendAuditEvent(
                 new AuditEvent(
                         auditEventType,
                         configService.getSsmParameter(ConfigurationVariable.COMPONENT_ID),
                         auditEventUser,
-                        new AuditExtensionCoiCheck(coiCheckType, coiCheckSuccess)));
+                        new AuditExtensionCoiCheck(coiCheckType, coiCheckSuccess),
+                        restrictedData));
+    }
+
+    private AuditRestrictedCheckCoi getRestrictedAuditDataforCheckCoi(
+            String userId, String ipvSessionId)
+            throws HttpResponseExceptionWithErrorBody, CredentialParseException,
+                    VerifiableCredentialException {
+        var oldVcs = verifiableCredentialService.getVcs(userId);
+        var sessionVcs = sessionCredentialsService.getCredentials(ipvSessionId, userId);
+
+        var oldIdentityClaim = userIdentityService.findIdentityClaim(oldVcs);
+        var sessionIdentityClaim = userIdentityService.findIdentityClaim(sessionVcs);
+
+        if (oldIdentityClaim.isEmpty() || sessionIdentityClaim.isEmpty()) {
+            LOGGER.error(LogHelper.buildLogMessage("Failed to generate identity claim"));
+            throw new HttpResponseExceptionWithErrorBody(
+                    500, ErrorResponse.FAILED_TO_GENERATE_IDENTIY_CLAIM);
```

Comment:
I would try and avoid throwing here - instead I'd prefer to log a warning and return null.